docs/src/man/doctests.md

```diff
@@ -105,59 +102,102 @@ line where checking should stop, i.e.
+
+````markdown
+```jldoctest mylabel
+julia> type Foo
+            x
+            y
+       end
+```
+
+Some text discussing the `Foo` type...
+In the next doctest block the type `Foo` can be used without redefining:
 
+```jldoctest mylabel
+julia> foo = Foo(1, 2)
+Foo(1,2)
+```
+
+Some more text...
+In the next block the variable `foo` can be used:
+
+```jldoctest mylabel
+julia> foo.x
+1
+
+julia> foo.y
+2
 ```
 ````
 
-## Skipping Doctests
+!!! note
 
-Doctesting can be disabled by setting the [`makedocs`](@ref) keyword `doctest = false`.
-This should only be done when initially laying out the structure of a package's
-documentation, after which it's encouraged to always run doctests when building docs.
+    Labeled doctest blocks does not need to be consecutive (as in the example above) to be
+    included in the same module. They can be interspaced with unlabeled blocks or blocks
+    with another label.
 
 ## Setup Code
 
 Doctests may require some setup code that must be evaluated prior to that of the actual
-example, but that should not be displayed in the final documentation. It could also be that
-several separate doctests require the same definitions. For both of these cases a `@meta`
-block containing a `DocTestSetup = ...` value can be used as follows:
+example, but that should not be displayed in the final documentation. For this purpose a
+`@meta` block containing a `DocTestSetup = ...` value can be used as follows:
 
-    ```jldoctest
-    julia> using DataFrames
-
-    julia> df = DataFrame(A = 1:10, B = 2:2:20);
+````markdown
+Some text discussing a function `foo`...
 
-    ```
+The definition of `foo` is defined in a `@meta` block below, and thus not visible in the
+rendered documentation:
 
-    Some text discussing `df`...
+```@meta
+DocTestSetup = quote
+    function foo(x)
+        return x^2
+    end
 
-    ```@meta
-    DocTestSetup = quote
-        using DataFrames
-        df = DataFrame(A = 1:10, B = 2:2:20)
+    function foo(x, y)
+        return x * y
     end
-    ```
-
-    ```jldoctest
-    julia> df[1, 1]
-    1
-    ```
-
-    Some more text...
-
-    ```jldoctest
-    julia> df[1, :]
-    1x2 DataFrames.DataFrame
-    | Row | A | B |
-    |-----|---|---|
-    | 1   | 1 | 2 |
-    ```
-
-    ```@meta
-    DocTestSetup = nothing
-    ```
-
-Note that the `DocTestSetup` value is **re-evaluated** at the start of *each* doctest block
-and no state is shared between any code blocks. The `DocTestSetup = nothing` is not strictly
-necessary, but good practice nonetheless to help avoid unintentional definitions later on a
-page.
+end
+```
+
+```jldoctest
+julia> foo(2)
+4
+```
+
+Some more text...
+
+```jldoctest
+julia> foo(2, 3)
+6
+```
+
+```@meta
+DocTestSetup = nothing
+```
+````
+
+!!! note
+
+    The `DocTestSetup` value is **re-evaluated** at the start of *each* doctest block
+    and no state is shared between any code blocks. The `DocTestSetup = nothing` is not strictly
```

Comment:
I started thinking that the no-state statement is not entirely accurate if you allow for labelled blocks, e.g. as in the following example:

````
```@meta
DocTestSetup = quote
    if isdefined(:foo)
        foo += 10
    else
        foo = 42
    end
end
```

```jldoctest
julia> foo += 1
43
```

```jldoctest
julia> foo += 1
43
```

```jldoctest label
julia> foo += 1
43
```

```jldoctest label
julia> foo += 1
54
```
````

But I am not sure if it is worth complicating the language here.